### PR TITLE
Feature/41

### LINKED ISSUE #10: Develop a configuration profile
# Main description

Design the profile for server that will contain required data.

# Possible profile

## Format

1. backendId
2. broker.host + broker.port
3. active

### LINKED ISSUE #13: Add a possibility to save information about devices, their sensors and data
# Main description

Implement class that will handle data from devices and their sensors. 

## Requirements

Store data (messages) which should be sent and could be got.

### Common

1. Create `Messages` interface in `Common/src/utils/protocol/messages.js`:
* `constructor()` -> init `collection` variable (`messages`)
* `function initialize() : void` -> initialize collection if it doesn't exist 
* `function get(message : stirng) : array of dictionaries with message data` -> get all data by message name
* `function save(message : string, data : dictionary with message data) : void` -> save data
2. Add interface into export of `index.js` for Common module

### Device (lowdb)

1. Implement Common interface (`Device/src/backend/protocol/messages.js`), similar code of work with lowdb is for ConfigurationProfile class (`Device/src/backend/profile/config.js`)
2. Location of file with messages: messages.json
3. Example of format:

```
{
    "MESSAGE1": [
        {
            "id": 1,  // autoincrement(1,1)
            "data": 
            {
                "value": 10,
                "time": 20181216
            }
        }
    ],
    "MESSAGE2": [ ... ]
}
```

### Server (MongoDB/Mongoose)

1. Implement Common interface (`Server/src/backend/protocol/messages.js`), similar code of work with MongoDB is for ConfigurationProfile class (`Server/src/backend/profile/config.js`)
2. Collection: messages
3. Schemas should describe `id` (numeric, autoincrement(1,1)) and `data` (dictionary, complex content)

### LINKED ISSUE #41: Introduce MongoDB
# Main description

Introduce MongoDB for Server module. It should collect data about **configuration profile** and **data from devices**.

### PR DESCRIPTION
**Pull request by boykosha**
Closes #41, closes #13, closes #10

### CODE REVIEW

File: Server/src/backend/protocol/messages.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        this.messageClass = await this.database.model(this.collection, messageSchema)
+        if ((await this.get()).length === 0) {
+            await this.messageClass.create({}).catch(err => {
+                logger.error(err)
+            })
+        }
+    }
+
+    async get(message) {
```
Comment (GUSAR1T0) [inline Server/src/backend/protocol/messages.js]: Function parameter `message` can be set by default, e.g.: `null`.

File: Server/src/backend/protocol/messages.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                logger.error(err)
+            })
+        }   
+    }
+
+    async save(message, data) {
+        var updatedData = []
+        if ((await this.get(message)).length !== 0) {            
+            var currentData = (await this.messageClass.findOne({message: message}, {data: 1, _id: 0}))
+            for (var i = 0; i < (currentData).data.length ; i++) {
```
Comment (GUSAR1T0) [inline Server/src/backend/protocol/messages.js]: Parentheses for `currentData` variable is redundant.

File: Server/src/backend/protocol/messages.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                updatedData.push((currentData).data[i])
+            }
+        }
+
+        updatedData.push(data)
+        
+        await this.messageClass.updateOne(
+            {message: `${message}`},
```
Comment (GUSAR1T0) [inline Server/src/backend/protocol/messages.js]: Format file, pls 😄

File: Device/src/backend/protocol/messages.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+const RegisterResponse = common.models.protocol.messages.RegisterResponse
+const SensorResponse = common.models.protocol.messages.SensorDataResponse
+
+class DeviceMessages extends Messages{
+    constructor() {
+        super()
+        this.database = databaseHandler(`${this.collection}.json`)
+    }
+
+    initialize() {
```
Comment (GUSAR1T0) [inline Device/src/backend/protocol/messages.js]: Could we use similar approach of initialization in lowdb as for MongoDB?

File: Device/src/backend/protocol/messages.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+const ActionDeviceRequest = common.models.protocol.messages.ActionDeviceRequest
+const ActionSensorRequest = common.models.protocol.messages.ActionSensorRequest
+const RegisterRequest = common.models.protocol.messages.RegisterRequest
+const ActionDeviceResponse = common.models.protocol.messages.ActionDeviceResponse
+const ActionSensorResponse = common.models.protocol.messages.ActionSensorResponse
+const RegisterResponse = common.models.protocol.messages.RegisterResponse
+const SensorResponse = common.models.protocol.messages.SensorDataResponse
+
+class DeviceMessages extends Messages{
```
Comment (GUSAR1T0) [inline Device/src/backend/protocol/messages.js]: Format file, pls

Comment (boykosha) [reply Device/src/backend/protocol/messages.js]: Done

File: Server/src/backend/protocol/messages.js (hunk 1 above)
Comment (boykosha) [reply Server/src/backend/protocol/messages.js]: Done

File: Server/src/backend/protocol/messages.js (hunk 2 above)
Comment (boykosha) [reply Server/src/backend/protocol/messages.js]: Removed

**Pull request merged**